src/app/view/perspectives/chart/model/collection-attribute-pair-packer.ts

```diff
@@ -0,0 +1,62 @@
+/*
+ * Lumeer: Modern Data Definition and Processing Platform
+ *
+ * Copyright (C) since 2017 Answer Institute, s.r.o. and/or its affiliates.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import {DocumentModel} from '../../../../core/store/documents/document.model';
+import {CollectionAttributePair} from './collection-attribute-pair';
+import {CollectionModel} from '../../../../core/store/collections/collection.model';
+
+type Pair = [CollectionModel, string];
+
+export class Packer {
+
+  public pack(documents: DocumentModel[]): CollectionAttributePair[] {
```

Comment:
This is exactly the Java style we discussed today. It's not a bad solution. But it would be much cleaner to move this method to `collection-attribute-pair.ts` and have it there as function called for example `createCollectionAttributePair`. The only exported function would be this one and the other private ones would not be exported and so they will not be visible from other modules.